substrate/frame/session/src/tests.rs

```diff
@@ -467,6 +467,91 @@ fn test_migration_v1() {
 	});
 }
 
+#[test]
+fn set_keys_should_fail_with_insufficient_funds() {
+	new_test_ext().execute_with(|| {
+		// Account 999 is mocked to have ExistentialDeposit but insufficient funds
+		let account_id = 999;
+		let keys = MockSessionKeys { dummy: UintAuthorityId(account_id).into() };
+		frame_system::Pallet::<Test>::inc_providers(&account_id);
+		// Make sure we have a validator ID
+		ValidatorAccounts::mutate(|m| {
```

Comment:
@Ank4n another follow-up here would be to also remove this, agreed? The author of this PR can do it as a next one :)